redis/typing.py

```diff
@@ -51,6 +53,36 @@
 ExceptionMappingT = Mapping[str, Union[Type[Exception], Mapping[str, Type[Exception]]]]
 
 
+# New typing work in progress
+
+BooleanType = bool
+IntegerType = int
+OptionalStringType = Optional[str]
+StringListType = List[str]
+OptionalStringListType = Optional[List[str]]
+
+ResponseTypeBoolean = TypeVar(
+    "ResponseTypeBoolean",
+    bound=Union[Awaitable[BooleanType], BooleanType],
```

Comment:
```suggestion
    bound=Union[Awaitable[bool], bool],
```

Reply:
> If we're not after 3.8 compatibility, then I think it should be

According to the [README](https://github.com/redis/redis-py/blob/6246cbade4fa9ae879455b498ef8fef5250619b1/README.md), the current version of this project (6.2.0) seems to support Python 3.9+ meaning the `list[str]` syntax above is usable, but the `bool | Awaitable[bool]` syntax would not be, as this was introduced in Python 3.10 ([source](https://docs.python.org/3/whatsnew/3.10.html))

Reply:
Right it was just changed recently https://github.com/redis/redis-py/commit/e5756daafaf80670b68454334ef00e924e70f07a it used to say >=3.8 before.